drivers/net/mlx5/mlx5_vdpa.c

```diff
@@ -147,14 +160,92 @@ mlx5_vdpa_query_virtio_caps(struct vdpa_priv *priv)
 	 * For now only set protocol features support
 	 */
 	priv->caps.max_num_virtqs = MLX5_VDPA_SW_MAX_VIRTQS_SUPPORTED;
-	priv->caps.virtio_net_features = (1ULL << VHOST_USER_F_PROTOCOL_FEATURES);
+	/*
+	 * TODO (shahafs): Take from QUERY HCA CAP Device Emulation Capabilities.
+	 * For now only set protocol features support
+	 */
+	priv->caps.notify_stride = MLX5_VDPA_SW_NOTIFY_STRIDE;
```

Comment:
Once you'll rebase, you'll see there's already piece of code that try to query the VIRTIO params from FW.
Plz Just add there as well to get the notify_stride (and in the else "FW no support" have this).
Also plz note the stride should come in a log from FW.

Reply:
Thanks. 

on the next version i removed the stride completely. the support for doorbell offset not page aligned is not yet ready for KVM. am planning to look into it. 

if we can make it work, will add the support.

Note that we consider to change ConnectX-6Dx doorbell layout due to it as well.